### PR TITLE
Airlock spawners

### PR DESCRIPTION
**Pull request by Citinited**
**What does this PR do:**

This PR adds a new tool for mappers, the airlock spawner. What this does is that it will, upon `Initialize`, spawn its own airlocks, control equipment, and any vents that are required for the airlock to function. This means that you no longer need to create your own airlocks and tediously VV everything - it'll do it all for you.

The spawner will not create walls, floors, or a canister for you, it will leave that down to you. Simply connect your piping to the indicated pipe on the airlock and the rest will be done for you!

You can specify any size and orientation of the generated chamber and resultant airlocks. You could specify the entrance to be in the north and the exit to be in the east, for example.

This PR also adds some helpers, namely `NORTH_OF_TURF(turf/T)`, `EAST_OF_TURF(turf/T)`, `SOUTH_OF_TURF(turf/T)`, and `WEST_OF_TURF(turf/T)`. These seemed like they could be useful elsewhere so I stuck them in the appropriate file.

**Map changes**

I haven't put these into deltastation, as I know someone is working on updating that and I don't want to conflict them. Metastation also doesn't have any cycling airlocks, so I left them off that.
There have been a couple minor map changes.

Security dorms: now has a 2 by 2 airlock instead of a a 1 by 2 airlock
image

Telecoms: now has two doors instead of just one.
image

**Sprites**

How it looks in the map editor: 
image

1 by 1 airlock
image

1 by 2 airlock
image

2 by 2 airlock
image

3 by 2 airlock
image

3 by 3 airlock
image

Here is a video of it working fine and dandy
https://streamable.com/3g1gb

:cl:
add: Mappers have a new tool that creates a fully functional cycling airlock.
tweak: Most cycling airlocks are now see-through. (Exceptions: syndicate shuttle, engine airlocks, and telecoms)
/:cl:

### CODE REVIEW

**Comment by datlo** [comment]
YES PLEASE

**Comment by FreeStylaLT** [comment]
img

**Comment by Spartan6** [comment]
Unf.

**Comment by Agameofscones** [comment]
For the better sprites maybe just loose outline the airlock shape and pipes in a nice stark ugly red so it's clear where yer' placin' yer objects? Also god bless yer soul mate, no more copying mining locks to every station bit

**Comment by Citinited** [comment]
@Agameofscones Yep that's what I've done, check out the sprites in the checks tab

I should rejig the sprites to show the airlocks though

**Comment by Citinited** [comment]
Hmm, I've just realised that n

**Comment by Citinited** [comment]
This is very nearly done, all I need to do is squash my commits and finish mapediting stuff and this will be review-ready

**Comment by taukausanake** [comment]
Quick question: Using this you can make any shape and size airlock and not just the ones that were defined in the file, correct? If I want to make a weird L airlock or U shape I could do that?

**Comment by Citinited** [comment]
The functionality for U-shaped airlocks isn't here yet (but I think it would be possible to code without too much difficulty), but L-shaped airlocks should be possible. See: the new telecoms airlock

**Comment by taukausanake** [comment]
I meant more like
```
[W] [A] [A] [W]
[W] [F] [F] [W] [W]
[W] [F] [F] [F] [A]
[W] [F] [F] [F] [A]
[W] [W] [W] [W] [W]
```
Where A is an airlock, F is the floor in an airlock, and W is a wall or such.
And a U shape taking a copy of that but flipping it to the side minus the airlocks on said flipped side. It's weird I know but I'm just curious before I do something stupid

**Comment by Citinited** [comment]
Neither of those are possible yet due to the way in which airlocks are generated (I didn't want to make it ridiculously complicated). However, the former is possible by adding a bit of code that adds another airlock if you want it (currently it only adds a second airlock if there are an even number of tiles to a side), and I'm not sure the latter is possible with the way I've got it set up currently. 

You'll have to map those in yourself if you want them, I'm afraid ;)

File: code/game/machinery/embedded_controller/airlock_controllers.dm [hunk 1]
```diff
@@ -14,7 +14,8 @@
 
 /obj/machinery/embedded_controller/radio/airlock/Initialize()
 	..()
-	program = new/datum/computer/file/embedded_program/airlock(src)
+	spawn(25) //Necessary as we initialize before we can use a timer. We need a bit of delay as otherwise the program may grab incorrect values
```
Comment (Citinited) [inline code/game/machinery/embedded_controller/airlock_controllers.dm]: Reminder for myself to check if this `spawn` is no longer needed

**Comment by variableundefined** [comment]
Inform when deconflicted

**Comment by Markolie** [comment]
This'd be amazing to have implemented prior to Lavaland, so we can also implement this on all the new maps being added!

**Comment by Citinited** [comment]
@Anasari I've deconflicted this (I hope), you can go ahead and reopen it.

**Comment by Markolie** [comment]
Would still love to have this in before Lavaland @variableundefined (and @Fox-McCloud @Crazylemon64 I guess). Could it be reopened?

**Comment by Fox-McCloud** [comment]
@Anasari

**Comment by Fox-McCloud** [comment]
@variableundefined

**Comment by variableundefined** [comment]
I cannot reopen it as it is force pushed or recreated, for some reason.

**Comment by taukausanake** [comment]
Maybe don't force close PRs then without the OP saying it's okay or not replying after declaring it stale? I honestly really hate seeing PRs just getting closed right after a declaration of any sort, good or bad

**Comment by Citinited** [comment]
Ah yes, I force pushed it in order to deconflict it. As this PR is presumably unsalvageable, I'll start a new PR.

**Comment by Fox-McCloud** [comment]
yeah, I can't open it either.

o7

**Pull request closed without merging**